File: pinyougou_parent/pinyougou_pay_interface/src/main/java/com/pinyougou/pay/service/PayService.java
package com.pinyougou.pay.service;

import com.pinyougou.pojo.TbPayLog;

import java.util.Map;

public interface PayService {
    public Map<String,Object> createNative(String out_trade_no,String total_fee) throws Exception;
    //查询支付状态的方法
    public Map<String,String> queryPayStatus(String out_trade_no) throws Exception;

    TbPayLog findPayLogByUserId(String userId);

    void updateStatus(String out_trade_no, String transaction_id);
}

File: pinyougou_parent/pinyougou_cart_web/src/main/webapp/js/service/payService.js
//服务层
app.service('payService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.createNative=function(){
		return $http.get('../pay/createNative.do');
	}
	//查询支付状态
	this.queryPayStatus=function (out_trade_no) {
        return $http.get('../pay/queryPayStatus.do?out_trade_no='+out_trade_no);
    }

});
